dataprofiler/profilers/profile_builder.py

```diff
@@ -259,6 +291,20 @@ def update_profile(self, df_series, sample_size=None,
             sample_size = self._get_sample_size(df_series)
         if not min_true_samples:
             min_true_samples = self._min_true_samples
+
+        if utils.overlap(self.min_id, self.max_id,
+                         min(df_series.index), max(df_series.index)):
+            # Increment df_series index so that no overlap with current data
+            warnings.warn("Overlapping indices detected between data given "
+                          "to update_profile and profiled data, indices of "
+                          "provided data will be shifted to resolve this.")
+            if isinstance(df_series.index, pd.RangeIndex):
+                new_start = df_series.index.start + self.max_id + 1
+                new_stop = df_series.index.stop + self.max_id + 1
+                df_series.index = pd.RangeIndex(start=new_start, stop=new_stop,
```

Comment:
realizing, will this change the external data passed in, unbeknownst to the user?

Reply:
^ I believe so, which is one of my concerns